File: eMovieFinder/eMovieFinder.Database/Entities/Order.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace eMovieFinder.Database.Entities
{
    public class Order
    {
        [Key]
        public int Id { get; set; }
        [ForeignKey("User")]
        public int UserId { get; set; }
        public virtual User User { get; set; }
        public DateTime OrderDate { get; set; }
        public virtual ICollection<OrderMovie> OrderMovies { get; set; }
    }
}

File: eMovieFinder-UI/emovie_finder_mobile/lib/src/models/entities/order.dart
import 'package:json_annotation/json_annotation.dart';

import 'package:emovie_finder_mobile/src/models/entities/user.dart';
import 'ordermovie.dart';

part 'order.g.dart';

@JsonSerializable()
class Order {
  final int? id;
  final int? userId;
  final DateTime? orderDate;
  final String? formattedOrderDate;
  final String? finalMoviePrice;
  final User? user;
  final List<OrderMovie>? orderMovies;

  Order({
    this.id,
    this.userId,
    this.orderDate,
    this.formattedOrderDate,
    this.finalMoviePrice,
    this.user,
    this.orderMovies
  });

  factory Order.fromJson(Map<String, dynamic> json) => _$OrderFromJson(json);
  Map<String, dynamic> toJson() => _$OrderToJson(this);
}